sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/base_node.py

```diff
@@ -204,6 +204,40 @@ def name(self, value):
             value = value.lower()
         self._name = value
 
+    def get_predecessors(self) -> List["BaseNode"]:
+        """Return list of predecessors for current node.
```

Comment:
Will the order be deterministic? If so, can we clarify this in docstring?

Reply:
The order is guaranteed, do we need to make it have some order?

Reply:
I think it depends on how to use it: if we just use this to check predecessor (e.g. `in xxx.get_predecessors()`), order is not matter at all; but if there is any advanced usage (I cannot imagine, so Det.Ch expected usage/scenario shall help me understand), order, or say knowing what is the order, is important.

Reply:
Just read Det.CH usage (from below comment), in that way, I think there is no need to make it some specific order. And if it's not easy to desribe the order, I think we can leave this out in docstring.

Reply:
Not sure if we need to mention this in docsting, since we don't have order for inputs/outputs, either.

Reply:
Yeah, it could be difficult in such case.